### PR TITLE
fix(avatar): Fix undesired halo around Dark variant

### PR DESCRIPTION
**Pull request by 6r3al**
## Summary

fix: #724 

## Checklist

- [ ] branch has been rebased on the latest master commit
- [ ] tests are changed or added
- [ ] `yarn test` passes
- [ ] all (dev)dependencies that the module needs is added to its `package.json`
- [ ] code has been documented and, if applicable, usage described in README.md
- [ ] module has been added to `canvas-kit-react` and/or `canvas-kit-css` universal modules, if
      applicable
- [ ] design approved final implementation
- [ ] a11y approved final implementation
- [ ] code adheres to the [API & Pattern guidelines](../API_PATTERN_GUIDELINES.md)

## Additional References

Before/after
Screenshot 2020-07-17 at 01 24 03

### CODE REVIEW

File: modules/avatar/react/lib/Avatar.tsx [hunk 1]
```diff
@@ -35,6 +35,11 @@ export interface AvatarProps extends React.ButtonHTMLAttributes<HTMLButtonElemen
    * Will render an `div` tag instead of a `button` when defined.
    */
   as?: 'div';
+  /**
+   * If true, render the image and hide the Avatar background.
+   * @default false
+   */
+  imageLoaded?: boolean;
```
Comment (NicholasBoll) [inline modules/avatar/react/lib/Avatar.tsx:42]: We don't want to add `imageLoaded` to the `AvatarProps` interface. It is only used by the `StyledContainer` and will be ignored by `Avatar`.

File: modules/avatar/react/lib/Avatar.tsx [hunk 2]
```diff
@@ -55,9 +60,8 @@ type AvatarOverload = {
 
 const StyledContainer = styled('button', {
   shouldForwardProp: prop => isPropValid(prop) && prop !== 'size',
-})<Pick<AvatarProps, 'size' | 'onClick'>>(
+})<Pick<AvatarProps, 'size' | 'onClick' | 'imageLoaded' | 'url'>>(
```
Comment (NicholasBoll) [inline modules/avatar/react/lib/Avatar.tsx:63]: Remove `imageLoaded` from the `AvatarProps` interface and make this change instead:

```suggestion
})<Pick<AvatarProps, 'size' | 'onClick' | 'url'> & { imageLoaded: boolean }>(
```

File: modules/avatar/react/lib/Avatar.tsx [hunk 3]
```diff
@@ -81,6 +85,9 @@ const StyledContainer = styled('button', {
   }),
   ({onClick}) => ({
     cursor: onClick ? 'pointer' : 'default',
+  }),
+  ({imageLoaded, url}) => ({
+    backgroundColor: !url || imageLoaded ? 'transparent' : colors.soap200,
```
Comment (NicholasBoll) [inline modules/avatar/react/lib/Avatar.tsx:90]: We don't want to transition `backgroundColor`. That requires a repaint in the browser. We want to transition `opacity` instead:

* `backgroundColor`: https://csstriggers.com/background-color
* `opacity`: https://csstriggers.com/opacity

**Comment by NicholasBoll** [comment]
This doesn't seem to fix the problem. I still see the halo on the story: https://5d854c26ba934e0020f5e98a-bmdvctujcl.chromatic.com/?path=/story/testing-react-indicators-avatar--avatar-states

Here's a screenshot:
Screen Shot 2020-07-17 at 12 35 25 PM

File: modules/avatar/react/lib/Avatar.tsx [hunk 4]
```diff
@@ -53,11 +53,12 @@ type AvatarOverload = {
   Size: typeof SystemIconCircleSize;
 };
 
+const fadeTransition = 'opacity 150ms linear';
+
 const StyledContainer = styled('button', {
   shouldForwardProp: prop => isPropValid(prop) && prop !== 'size',
 })<Pick<AvatarProps, 'size' | 'onClick'>>(
   {
-    background: colors.soap200,
```
Comment (jamesfan) [inline modules/avatar/react/lib/Avatar.tsx:60]: Forgot to add this `background` back in.

**Pull request merged**